packages/cli/generators/datasource/templates/datasource.ts.ejs

```diff
@@ -1,7 +1,8 @@
-import {inject} from '@loopback/core';
+import {inject, lifeCycleObserver} from '@loopback/core';
 import {juggler} from '@loopback/repository';
 import * as config from './<%= jsonFileName %>';
 
+@lifeCycleObserver('datasource')
 export class <%= className %>DataSource extends juggler.DataSource {
```

Comment:
should the class also implement LifeCycleObserver?
```typescript
export class <%= className %>DataSource extends juggler.DataSource implements LifeCycleObserver {
```

Reply:
+1

Reply:
It's not required to be functional but we can add it to be explicit.

Reply:
By adding `implements LifeCycleObserer`, we allow TypeScript compiler to catch certain kind of errors and also make future refactorings easier because of the extra type information.